test/e2e/framework/kubelet/config.go

```diff
@@ -30,11 +32,65 @@ import (
 	kubeletconfigv1beta1 "k8s.io/kubelet/config/v1beta1"
 	kubeletconfig "k8s.io/kubernetes/pkg/kubelet/apis/config"
 	kubeletconfigscheme "k8s.io/kubernetes/pkg/kubelet/apis/config/scheme"
-
 	"k8s.io/kubernetes/test/e2e/framework"
 	e2ekubectl "k8s.io/kubernetes/test/e2e/framework/kubectl"
+
+	"sigs.k8s.io/yaml"
 )
 
+func getKubeletConfigFilePath() (string, error) {
+	cwd, err := os.Getwd()
+	if err != nil {
+		return "", fmt.Errorf("failed to get current working directory: %v", err)
+	}
+
+	// DO NOT name this file "kubelet" - you will overwrite the kubelet binary and be very confused :)
+	return filepath.Join(cwd, "kubelet-config"), nil
+}
+
+// GetCurrentKubeletConfigFromFile returns the current kubelet configuration specified under /var/lib/kubelet/config.yaml.
+// This method should only run together with e2e node tests, meaning the test executor and the cluster nodes is the
+// same machine
+func GetCurrentKubeletConfigFromFile() (*kubeletconfigv1beta1.KubeletConfiguration, error) {
+	kubeletConfigFilePath, err := getKubeletConfigFilePath()
+	if err != nil {
+		return nil, err
+	}
+
+	data, err := ioutil.ReadFile(kubeletConfigFilePath)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get the kubelet config from the file %q: %v", kubeletConfigFilePath, err)
+	}
+
+	var kubeletConfig kubeletconfigv1beta1.KubeletConfiguration
+	if err := yaml.Unmarshal(data, &kubeletConfig); err != nil {
+		return nil, fmt.Errorf("failed to unmarshal the kubelet config: %v", err)
+	}
+
+	return &kubeletConfig, nil
+}
+
+// UpdateKubeletConfigFile updates the kubelet configuration under the filesystem
+// This method should only run together with e2e node tests, meaning the test executor and the cluster nodes is the
+// same machine
+func UpdateKubeletConfigFile(kubeletConfig *kubeletconfigv1beta1.KubeletConfiguration) error {
```

Comment:
I'd suggest to call it `WriteKubeletConfigFile`. As this method is not restarting kubelet, just writing the file. "Update" may cause confusion. wdyt?

Reply:
Works for me. Done